textscreen/txt_window.c

```diff
@@ -546,8 +547,12 @@ void TXT_OpenURL(char *url)
     TXT_snprintf(cmd, cmd_len, "xdg-open \"%s\"", url);
 #endif
 
-    system(cmd);
+    retval = system(cmd);
     free(cmd);
+    if (retval != 0) {
+        fprintf(stderr,
+                "executing xdg-utils failed, return value %d. Can't open this URL:\n%s\n", retval, url);
```

Comment:
On MacOS, xdg-utils isn't being used, so this becomes a misleading message. I suggest the following:


    retval = system(cmd);
    if (retval != 0)
    {
        fprintf(stderr, "TXT_OpenURL: error executing '%s'; return code %d\n",
                cmd, retval);
    }

    free(cmd);